samples/extensions/mesh_shader_culling/README.md

```diff
@@ -0,0 +1,188 @@
+taskPayloadSharedEXT SharedData sharedData;
+```
+
+Once a variable is defined with taskPayloadSharedEXT in the task shader, it will be shared with the mesh shader when 
+EmitMeshTasksEXT is called.
+
+In general, the Mesh pipeline refers to a new pipeline 
+which replaces everything before the fragment shader 
+with an (optional) task shading stage that can call other mesh 
+shading stages.
+
+* A task shader (optional) is used to emit mesh shader(s)
+* A mesh shader has the responsibility to generate 
+  primitives.
+
+More details about emitting a mesh task can be found in the attached article:
+
+[Introduction to Turing Mesh Shaders](https://developer.nvidia.com/blog/introduction-turing-mesh-shaders/)
+
+GPU manufactures have recommended best practices for 
+their hardware in setting the work group and mesh size 
+number. Further reading can be found here:
+
+* [Meshlet Size tradeoffs](https://zeux.io/2023/01/16/meshlet-size-tradeoffs/)
+* NVIDIA - [Mesh Shaders in Turing](https://on-demand.gputechconf.com/gtc-eu/2018/pdf/e8515-mesh-shaders-in-turing.pdf)
+* AMD - [Sampler feedback ultimate in Mesh shaders](https://gpuopen.com/wp-content/uploads/slides/AMD_RDNA2_DirectX12_Ultimate_SamplerFeedbackMeshShaders.pdf)
+* [Timur's blog](https://timur.hu/blog/2022/mesh-and-task-shaders)
+
+## Mesh Shader
+
+A mesh shader executes on the same workgroup as the task shader that emitted it if a task shader exists. A 
+mesh shader is responsible for generating 
+vertices and indices 
+based on the number of meshlets determined by its task
+shader. The vertices and indices generation process can be 
+found in the following code:
+
+```glsl
+// Vertices:
+gl_MeshVerticesEXT[k * 4 + 0].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_0, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 1].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_1, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 2].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_2, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 3].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_3, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+// Indices
+gl_PrimitiveTriangleIndicesEXT[k * 2 + 0] = unitPrimitive_0 + k * uvec3(4);
+gl_PrimitiveTriangleIndicesEXT[k * 2 + 1] = unitPrimitive_1 + k * uvec3(4);
+// Assigning the color output:
+vec3 color = vec3(1.0f, 0.0f, 0.0f) * (k + 1) / sharedData.meshletsNumber;
+outColor[k * 4 + 0] = color;
+outColor[k * 4 + 1] = color;
+outColor[k * 4 + 2] = color;
+outColor[k * 4 + 3] = color;
+```
+
+More details of meshlets generation can be found in the attached article:
+
+[Using Mesh Shaders for Professional Graphics](https://developer.nvidia.com/blog/using-mesh-shaders-for-professional-graphics/)
+
+## Meshlets culling
```

Comment:
What is described below is not meshlet culling but per-primitive culling.

To clarify:

* Per-meshlet culling is done in the task shader. One example of how this can be done is having a bounding box of the entire meshlet and test whether the entire meshlet would be invisible. This is highly recommended because it reduces the number of mesh shader workgroups launched by that shader, so it saves compute (and bandwidth) resources by not launching any mesh shader workgroups for meshlets that would be invisible.
* Per-primitive culling is done in the mesh shader and its purpose is to have the mesh shader emit fewer vertices and primitives. This is typicaly less beneficial and is only recommended for scenes with very high geometric complexity or on GPUs whose fixed-function culling is slower than shaders. This is no longer recommended by neither AMD nor NVidia for their latest generation GPUs.